### PR TITLE
Add section to README about project naming history

### PR DESCRIPTION
**Pull request by mjankowski**
This is an attempt to address the concerns raised in https://github.com/thoughtbot/factory_girl/issues/921 and https://github.com/thoughtbot/factory_girl/issues/849 without a full project rename.

### CODE REVIEW

**Comment by croaky** [comment]
Duplicate of https://github.com/thoughtbot/factory_girl/pull/942?

How about a link from the README to a `NAME.md`?

**Comment by mjankowski** [comment]
I saw #942 immediately after creating this, and send my regards to that PR.

Is your suggestion to move out the whole explanation to a new file so that we reduce the text being added to README, but make sure we link (earlyish in the file) to that naming explanation from the README?

File: README.md [hunk 1]
```diff
@@ -15,6 +15,19 @@ have [a detailed introductory video][], available for free on Upcase.
 
 [a detailed introductory video]: https://upcase.com/videos/factory-girl?utm_source=github&utm_medium=open-source&utm_campaign=factory-girl
 
+Project History and Naming
+--------------------------
+
+The name "Factory Girl" might be confusing to some developers who encounter this library. The name [was chosen](https://robots.thoughtbot.com/waiting-for-a-factory-girl) as a clever (or so we hoped) reference to the [Rolling Stones song](https://www.youtube.com/watch?v=4jKix2DFlnA) of the same name, and as a nod in the direction of the [Factory method pattern](https://en.wikipedia.org/wiki/Factory_method_pattern) from the Design Patterns book. Our intentions were somewhere between amusing and harmless, and any meaning beyond that which can be attached to the project name is unintentional.
```
Comment (croaky) [inline README.md]: "Our intentions were somewhere between amusing and harmless, and " - delete, repeats "he name [was chosen](https://robots.thoughtbot.com/waiting-for-a-factory-girl) as a clever"

Comment (croaky) [inline README.md]: "as a clever (or so we hoped)" -> "as a"

File: README.md [hunk 2]
```diff
@@ -15,6 +15,19 @@ have [a detailed introductory video][], available for free on Upcase.
...
+
+Through the power of revisionist history, we are comfortable with claims that the project is named after any of:
+
+- [Rosie the Riveter](https://en.wikipedia.org/wiki/Rosie_the_Riveter), a symbol of feminism and female economic power
+- [Factory Girls](https://www.youtube.com/watch?v=lHKNPWXaRZ8), a song by Flogging Molly
+- [Factory Girl](https://en.wikipedia.org/wiki/Factory_Girl_(film)), a film about Edie Sedgwick and her relationship with artist Andy Warhol
+
+We believe that the hassle and churn (and more confusion!) that would be caused by renaming this project outweighs the hypothetical clarity benefit that might come from doing that, and so we plan on keeping the name in place.
```
Comment (croaky) [inline README.md]: "the hassle and churn (and more confusion!)" -> "the added confusion and burden to maintainers and users"

"clarity benefit" -> "clarity"

**Comment by croaky** [comment]
> Is your suggestion to move out the whole explanation to a new file so that we reduce the text being added to README, but make sure we link (earlyish in the file) to that naming explanation from the README?

Yup

File: NAME.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Project Name and History
+------------------------
+
+The name "Factory Girl" might be confusing to some developers who encounter this
+library. The name [was chosen](https://robots.thoughtbot.com/waiting-for-a-factory-girl)
```
Comment (wardpenney) [inline NAME.md]: Passive voice `was chosen`. How about, `we chose`?

File: README.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ factory_girl is a fixtures replacement with a straightforward definition syntax,
 If you want to use factory_girl with Rails, see
 [factory_girl_rails](https://github.com/thoughtbot/factory_girl_rails).
 
+[Confused by the project name?](NAME.md)
```
Comment (wardpenney) [inline README.md]: How about `Interested in the project name?`

File: NAME.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- [Rosie the Riveter](https://en.wikipedia.org/wiki/Rosie_the_Riveter), a symbol
+  of feminism and female economic power
+- [Factory Girls](https://www.youtube.com/watch?v=lHKNPWXaRZ8), a song by
+  Flogging Molly
+- [Factory Girl](https://en.wikipedia.org/wiki/Factory_Girl_(film)), a film
+  about Edie Sedgwick and her relationship with artist Andy Warhol
+
+We believe that the added confusion and burden to maintainers and users that
+would be caused by renaming this project outweighs the hypothetical clarity that
```
Comment (wardpenney) [inline NAME.md]: I am not sure this paragraph is needed.

File: NAME.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  Flogging Molly.
+- [Factory Girl](https://en.wikipedia.org/wiki/Factory_Girl_(film)), a film
+  about Edie Sedgwick and her relationship with artist Andy Warhol.
+
+## Future of the name
+
+We believe that the added confusion and burden to maintainers and users that
+would be caused by renaming this project outweighs the hypothetical clarity that
+might be gained from changing the name. Given that, we plan on keeping the name
+in place for now, and will re-visit this decision from time to time.
```
Comment (croaky) [inline NAME.md]: Could probably delete this section for this purpose of this PR.

**Comment by mjankowski** [comment]
Made a new file, reduced some verbiage, squashed, probably acceptable for a first pass thing which can be expanded later.

File: NAME.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# Why is this project called "Factory Girl"?
+
+The name "Factory Girl" might be confusing to some developers who encounter this
+library.
+
+[We chose the name](https://robots.thoughtbot.com/waiting-for-a-factory-girl)
+as a reference to the [Rolling Stones song](https://www.youtube.com/watch?v=4jKix2DFlnA)
```
Comment (croaky) [inline NAME.md]: How about putting the technical references first? Feels like the initial reason, and Rolling Stones second.

File: NAME.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+of the same name, and as a nod in the direction of the [Factory method](https://en.wikipedia.org/wiki/Factory_method_pattern)
+and [Object Mother](http://martinfowler.com/bliki/ObjectMother.html) patterns
+from the _Design Patterns_ book. Any meaning beyond that which can be attached to
+the project name is unintentional.
```
Comment (croaky) [inline NAME.md]: I'm not loving this last sentence without further context.

**Comment by joshuaclayton** [comment]
Thanks @mjankowski; merged as 903d4325feb9c373863601cdab3fca4b6118fca7

**Pull request closed without merging**